src/wireguard_tools/wireguard_netlink.py

```diff
@@ -43,11 +44,18 @@ def get_config(self) -> WireguardConfig:
             listen_port=attrs["WGDEVICE_A_LISTEN_PORT"] or None,
         )
 
+        peer_attrs_by_pubkey: defaultdict[str, dict[str, Any]] = defaultdict(dict)
```

Comment:
this can probably be `defaultdict[bytes, dict[str, Any]]` so we can skip the extra decode("utf-8") calls as we iterate through the netlink results and recombine peers.